packages/slice-machine/src/features/onboarding/useOnboarding.ts

```diff
@@ -0,0 +1,62 @@
+import { OnboardingState } from "@prismicio/editor-fields";
+import { useRefGetter } from "@prismicio/editor-support/React";
+import { updateData, useRequest } from "@prismicio/editor-support/Suspense";
+import { toast } from "react-toastify";
+
+import { managerClient } from "@/managerClient";
+
+const { fetchOnboarding, toggleOnboarding, toggleOnboardingStep } =
+  managerClient.prismicRepository;
+
+async function getOnboarding() {
+  try {
+    return fetchOnboarding();
+  } catch (error) {
+    console.error("Failed to fetch onboarding", error);
+    return undefined;
+  }
+}
+
+export function useOnboarding() {
+  const onboarding = useRequest(getOnboarding, []);
+  const getOnboardingState = useRefGetter(onboarding);
+
+  function updateCache(onboarding: OnboardingState) {
```

Comment:
💡 
```suggestion
  function updateCache(newOnboardingState: OnboardingState) {
```

A bit hard to read when it has the same name as a parent `onboarding` var.

Reply:
Fine by me, renamed ✅